src/persist-client/src/fetch.rs

```diff
@@ -778,7 +767,8 @@ impl Cursor {
                 }
             };
 
-            let t = T::decode(t);
+            let mut t = T::decode(t);
+            t.advance_by(encoded.ts_rewrite.borrow());
```

Comment:
I don't have a good intuition for whether truncation or advancement ought to happen first. I think it's a moot point for now since no batch should ever be written using both features... but please correct me if that's not right!

Reply:
I thought about this, but forgot to write a comment. I have a strong intuition that advancement should come before truncation, but I'm not sure I have a convincing argument for why yet.

> since no batch should ever be written using both features... but please correct me if that's not right!
Ah interesting, I don't think there is anything that would disallow that. I think we'd have to add a check for it. Something like batch upper and CaA upper must be equal and `min(batch lower, batch rewrite_ts)` and CaA lower must be equal?

Reply:
Ah, I meant more that "in the forseeable future" nobody needs both, so if we regret our choice we can swap it around without too much suffering.

But codifying that in an assert sounds reasonable if it's straightforward to do!

Reply:
Added the assert and a comment